ghost/core/core/server/web/parent/middleware/queue-request.js

```diff
@@ -0,0 +1,76 @@
+const debug = require('@tryghost/debug')('middleware:queue-request');
+const metrics = require('@tryghost/metrics');
+const path = require('node:path');
+const expressQueue = require('express-queue');
+
+const DEFAULT_QUEUE_LIMIT = 100;
+
+module.exports = function queueRequest(
+    config,
+    queueFactory = expressQueue,
+    logMetric = metrics.metric.bind(metrics)
+) {
+    config = {
+        ...{
+            limit: DEFAULT_QUEUE_LIMIT
+        },
+        ...config
+    };
+
+    debug('Initialising middleware with config:', config);
+
+    // @see https://github.com/alykoshin/express-queue#usage
+    const queue = queueFactory({
+        activeLimit: config.limit,
+        queuedLimit: -1 // Do not limit the number of queued requests
```

Comment:
Can we think of a reason to need a config value for `queuedLimit`? I think we're ok with it kept at -1 because other parts of Pro have rate limits 🤔.

Reply:
🤔 Could potentially reduce memory bloat under heavy load (not having to store the reqs in the queue)

If this value is set we can provide a handler that gets executed (i.e throw a `503` or whatever)

tbh I think it should be ok as is, can always tweak if needed 👍

Reply:
`express-queue` should throw a 503 by default. Sounds good though.